### PR TITLE
Align module_db compatibility for XPU

### PR DESCRIPTION
**Pull request by PenghuiCheng**
module_db is used in test_modules.py.
1. There are decorators in the Module_info in module_db for CUDA device, so we align the XPU device with CUDA.
2. Some onInfo doesn't have a member named `skip`.

### CODE REVIEW

File: test/xpu/xpu_test_utils.py [hunk 1]
```diff
@@ -514,28 +515,30 @@ def replace_opinfo_device(self, info: OpInfo):
...
 
     def adjust_db(self, db):
         for opinfo in db:
-            if opinfo.name not in _xpu_computation_op_list:
-                opinfo.dtypesIfXPU = opinfo.dtypes
-            else:
-                backward_dtypes = set(opinfo.backward_dtypesIfCUDA)
-                backward_dtypes.add(bfloat16)
-                opinfo.backward_dtypes = tuple(backward_dtypes)
+            if not isinstance(opinfo, ModuleInfo):
```
Comment (fengyuan14) [inline test/xpu/xpu_test_utils.py:535]: Why don't use our supported list?

File: test/xpu/xpu_test_utils.py [hunk 2]
```diff
@@ -514,28 +515,30 @@ def replace_opinfo_device(self, info: OpInfo):
             info.decorators = tuple(decorator_xpu)
         skip_xpu = []
         replaced = False
-        for skip in info.skips:
-            if type(skip) == DecorateInfo:
-                if skip.device_type == "cuda":
-                    skip_xpu.append(decorator)
-                    skip.device_type = "xpu"
+        if hasattr(info, "skips"):
```
Comment (fengyuan14) [inline test/xpu/xpu_test_utils.py:518]: You are saying there is no `skips` in some `onInfo`?

File: test/xpu/xpu_test_utils.py (hunk 1 above)
Comment (fengyuan14) [inline test/xpu/xpu_test_utils.py:535]: You are saying ops in `ModuleInfo` are not listed in our supported list? So,
1. What operators are included in `ModuleInfo`?
2. Looks no operator in `ModuleInfo` is supported by us?

File: test/xpu/xpu_test_utils.py (hunk 2 above)
Comment (yuchengliu1) [reply test/xpu/xpu_test_utils.py:518]: Class `ModuleInfo` has the number `skips`. `hasattr` is not necessary.

Comment (PenghuiCheng) [reply test/xpu/xpu_test_utils.py:518]: There is no skips in ModuleInfo.

File: test/xpu/xpu_test_utils.py [hunk 3]
```diff
@@ -514,28 +515,30 @@ def replace_opinfo_device(self, info: OpInfo):
...
-                backward_dtypes = set(opinfo.backward_dtypesIfCUDA)
-                backward_dtypes.add(bfloat16)
-                opinfo.backward_dtypes = tuple(backward_dtypes)
+            if not isinstance(opinfo, ModuleInfo):
+                if opinfo.name not in _xpu_computation_op_list:
+                    opinfo.dtypesIfXPU = opinfo.dtypes
+                else:
+                    backward_dtypes = set(opinfo.backward_dtypesIfCUDA)
+                    backward_dtypes.add(bfloat16)
+                    opinfo.backward_dtypes = tuple(backward_dtypes)
```
Comment (fengyuan14) [inline test/xpu/xpu_test_utils.py:541]: Please wrapper code piece into a function, like `align_supported_dtypes`. So the logic of `adjust_db` is like,
1. `align_supported_dtypes`
2. `align_skips`

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**